samples/DotNet/MSI/EventHubsMSIDemoWebApp/EventHubsMSIDemoWebApp/EventHubsMSIDemo.aspx.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.ServiceBus;
+using Microsoft.ServiceBus.Messaging;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+// Always add app to IAM roles
+// Don't use on deployment slots but only on root
+namespace EventHubsMSIDemoWebApp
+{
+    public partial class EventHubsMSIDemo : System.Web.UI.Page
+    {
+        protected void Page_Load(object sender, EventArgs e)
+        {
+
+        }
+
+        protected void btnSend_Click(object sender, EventArgs e)
+        {
+            MessagingFactorySettings messagingFactorySettings = new MessagingFactorySettings
+            {
+                TokenProvider = TokenProvider.CreateManagedServiceIdentityTokenProvider(ServiceAudience.EventHubsAudience),
+                TransportType = TransportType.Amqp
+            };
+
+            // TODO - Remove after backend is patched with the AuthComponent open fix
+            messagingFactorySettings.AmqpTransportSettings.EnableLinkRedirect = false;
+
+            MessagingFactory messagingFactory = MessagingFactory.Create($"sb://{txtNamespace.Text}.servicebus.windows.net/",
+                messagingFactorySettings);
+
+            EventHubClient ehClient = messagingFactory.CreateEventHubClient(txtEventHub.Text);
+            ehClient.Send(new EventData(Encoding.UTF8.GetBytes(txtData.Text)));
+            ehClient.Close();
+            messagingFactory.Close();
+        }
+
+        protected void btnReceive_Click(object sender, EventArgs e)
+        {
+            MessagingFactorySettings messagingFactorySettings = new MessagingFactorySettings
+            {
+                TokenProvider = TokenProvider.CreateManagedServiceIdentityTokenProvider(ServiceAudience.EventHubsAudience),
+                TransportType = TransportType.Amqp
+            };
+
+            messagingFactorySettings.AmqpTransportSettings.EnableLinkRedirect = false;
+
+            MessagingFactory messagingFactory = MessagingFactory.Create($"sb://{txtNamespace.Text}.servicebus.windows.net/",
+                messagingFactorySettings);
+
+            EventHubClient ehClient = messagingFactory.CreateEventHubClient(txtEventHub.Text);
+            EventHubConsumerGroup consumerGroup = ehClient.GetDefaultConsumerGroup();
+            int partitions = int.Parse(txtPartitions.Text);
+            string[] Offsets = new string[partitions];
+            if (!string.IsNullOrEmpty(hiddenStartingOffset.Value))
```

Comment:
>hiddenStartingOffset [](start = 38, length = 20)

When page is refreshed this logic will be broken and it will be hard to tell user this on the page. Isn't it better to read from the start of the stream or say last 5 minutes?